src/mds/MDLog.cc

```diff
@@ -605,11 +614,29 @@ void MDLog::try_to_commit_open_file_table(uint64_t last_seq)
   }
 }
 
-void MDLog::trim(int m)
+void MDLog::log_trim_upkeep(void) {
+  dout(10) << dendl;
+
+  std::unique_lock mds_lock{mds->mds_lock};
+  while (!upkeep_log_trim_shutdown.load()) {
+    if (mds->is_active() || mds->is_stopping()) {
+      trim();
+    }
+
+    auto now = ceph::coarse_mono_clock::now();
+    auto when = now +
+      ceph::make_timespan(g_conf().get_val<std::chrono::seconds>("mds_log_trim_upkeep_interval").count());
```

Comment:
This is a little confusing, since `std::chrono::seconds` keeps the count as `int64_t`, while `ceph::make_timespan` expects a `double`. 
I like that this unambiguously conveys our expectation of the setting to be in seconds, however, one could incorrectly assume that changing the setting to `type: float` would work seamlessly.
Not suggesting a paricular edit here, just expressing my opinion.

Reply:
This has been simplified now (yet to push).